libbpf-tools/cpudist.bpf.c

```diff
@@ -71,8 +71,7 @@ static __always_inline void update_hist(struct task_struct *task,
 		histp = bpf_map_lookup_elem(&hists, &id);
 		if (!histp)
 			return;
-		bpf_probe_read_kernel_str(&histp->comm, sizeof(task->comm),
-					  task->comm);
+		BPF_CORE_READ_STR_INTO(&histp->comm, task, comm);
```

Comment:
this and below BPF_CORE_READ will actually call kernel helpers instead of direct memory access. But I understand this is necessary for raw_tp. I think it is okay for portability perspective. We have quite a few tools still using kprobe which uses BPF_CORE_READ as well.

Reply:
Thanks for the review, do you mean this will have a latency cost to use BPF_CORE_READ instead of direct mem access? Maybe another way is to separate each other with independent functions containing the same logic?